homeassistant/helpers/device_registry.py

```diff
@@ -958,16 +958,6 @@ def async_update_device(  # noqa: C901
             new_values["config_entries"] = config_entries
             old_values["config_entries"] = old.config_entries
 
-        for attr_name, setvalue in (
-            ("connections", merge_connections),
-            ("identifiers", merge_identifiers),
-        ):
-            old_value = getattr(old, attr_name)
-            # If not undefined, check if `value` contains new items.
-            if setvalue is not UNDEFINED and not setvalue.issubset(old_value):
-                new_values[attr_name] = old_value | setvalue
-                old_values[attr_name] = old_value
-
```

Comment:
I can see some code paths (looking within this function), where the only place (e.g.) `new_values['connections']` is  assigned is within the block you are removing.

Please can you help explain why not setting 'connections' in those code paths is ok?

Reply:
Can you  be more explicit, please? Which code paths exactly do you mean?

Reply:
If `async_update_device()` is called with `merge_connections`, `merge_identifiers`, `new_connections`, `new_identifiers` all `UNDEFINED`, then `new_values["connections"]` and `new_values["identifiers"]` won't be assigned to anything.

Reply:
Which is exactly how it was before. Nothing changes in that case.

Reply:
Ok, yes i see now. Thanks!